plugins/extractors/maxcompute/maxcompute.go

```diff
@@ -253,14 +255,16 @@ func (e *Extractor) buildTableAttributesData(schemaName string, tableInfo *table
 		attributesData["sql"] = tableInfo.ViewText
 	}
 
+	partitionNames := make([]string, len(tableInfo.PartitionColumns))
 	if tableInfo.PartitionColumns != nil && len(tableInfo.PartitionColumns) > 0 {
-		partitionNames := make([]string, len(tableInfo.PartitionColumns))
 		for i, column := range tableInfo.PartitionColumns {
 			partitionNames[i] = column.Name
 		}
 		attributesData["partition_fields"] = partitionNames
 	}
 
+	attributesData["max_preview_rows"] = e.client.MaxPreviewRows(table, "", int64(e.config.MaxPreviewRows))
```

Comment:
this is not the expected metadata we are looking for preview rows, please take a look the example from bigquery here

https://github.com/goto/meteor/blob/1eaeb72bce131f2db52dca0c3a9189cc11c9c5e0/plugins/extractors/bigquery/bigquery.go#L430-L438

https://github.com/goto/meteor/blob/1eaeb72bce131f2db52dca0c3a9189cc11c9c5e0/plugins/extractors/bigquery/bigquery.go#L477-L482